### DISCUSSION TITLE
multi-cloud setup according to docs, flannel wireguard backend shows internal ip's having the external ips

### DISCUSSION THREAD

**Discussion by migs35323** [k3s-io/k3s#7678] [General]
**Environmental Info:**
K3s Version: 1.24.10

Node(s) CPU architecture, OS, and Version: 
ubuntu 22

Cluster Configuration:
1 server 2 workers

**Describe the bug:**

hello, i was testing a new vps cluster and i come across this documentation: https://docs.k3s.io/installation/network-options#distributed-hybrid-or-multicloud-cluster

is it normal that the internal and external ip addresses are the same?
the nodes show having a flannel-wg interface..
i did install wireguard as instructed, do i need to create a wireguard interface or config ?
how do i can know flannel is indeed using wireguard and the packets are going trough wireguard?

**Steps To Reproduce:**

- install wireguard "apt install wireguard" (following https://docs.k3s.io/installation/network-options)
- use multicloud instructions to install a new cluster (https://docs.k3s.io/installation/network-options#distributed-hybrid-or-multicloud-cluster)

**behavior:**

the cluster assumes the external ip as the internal ip, 
(meaning both private and public ip are the vps external ip)
i am not sure if this is the normal behavior, or it should assume a private network

**Comment by rbrtbnfgl** [comment]
Hi @migs35323 
`flannel-wg` is the interface created by Flannel when it's using wireguard. If you installed the wireguard tool you can use the tool `wg show` to check the tunnel configurations. If you didn't specify any `internal-ip` or `external-ip` on the configuration K3s should use the first defaultGW IP for both.

**Comment by migs35323** [comment]
hello @rbrtbnfgl 
thanks for the tip, when using wg show it indeed shows the 2 other nodes
heres more info:
My VPS provider doesn't have internal network by default and even if it had, it wouldn't allow me to connect to nodes from other providers, so the solution in the documentation is perfect for me 
( https://docs.k3s.io/installation/network-options#distributed-hybrid-or-multicloud-cluster )

i ONLY provided the external ip as mentioned in the doc, i didn't provide an internal IP, i assumed the k3s would take care of that automatically

**Comment by caroline-suse-rancher** [comment]
This seems like more of a line of questioning than a bug, so converting to a discussion :)
